scipy/optimize/_linprog_util.py

```diff
@@ -69,6 +69,58 @@ def _check_sparse_inputs(options, A_ub, A_eq):
     return options, A_ub, A_eq
 
 
+def _format_A_constraints(A, n_x, sparse_lhs=False):
```

Comment:
This does seem to do the trick. For readability, though, consider a solution that is just one line longer:
```
if A is None and sparse_lhs:
    return sps.coo_matrix((0, n_x), dtype=float)
elif A is None and not sparse_lhs:
    return np.zeros((0, n_x), dtype=float)
elif A is not None and sparse_lhs:
    return sps.coo_matrix(A, dtype=float, copy=True)
else:
    return np.array(A, dtype=float, copy=True)
```

I think nested if would be even more readable, but some might disagree.

Reply:
I've changed it to use the additional ``elif`` line. 

This was original supposed to serve as a function within ``_clean_inputs(..)`` however I realised that just because ``A_eq is None`` does not mean ``A_ub is None`` so moved it to a new function.

I think avoiding nested if statements makes the logic clearer - we're trying to decide on the same thing, so it's written on the same level.

Reply:
I think it is a bit too convoluted

```python
if sparse_lhs:
    return sps.coo_matrix((0, n_x) if A is None else A, dtype=float)
else:
    if A is None:
        return np.zeros((0, n_x), dtype=float) 
    else:
        return np.array(A, dtype=float, copy=True)
```
seems well separated since first one decides whether sparsity involved and the next is checking Noneness of A

Reply:
@ilayn that looks good! I hadn't thought of putting the ternary into the matrix conversion.

Reply:
Yes, and I like the nested `if` : )